pallets/system/src/header_builder/version_select.rs

```diff
@@ -20,6 +20,66 @@ use std::{sync::OnceLock, time::Instant, vec::Vec};
 
 static PMP: OnceLock<M1NoPrecomp> = OnceLock::new();
 
+#[cfg(feature = "std")]
+pub fn get_empty_header(data_root: H256, version: HeaderVersion) -> HeaderExtension {
+	use avail_core::DataLookup;
+	let empty_commitment: Vec<u8> = vec![0];
+	let empty_app_lookup = DataLookup::new_empty();
+
+	match version {
+		HeaderVersion::V3 => {
+			let commitment = kc::v3::KateCommitment::new(1, 4, data_root, empty_commitment);
+			he::v3::HeaderExtension {
+				app_lookup: empty_app_lookup,
+				commitment,
+			}
+			.into()
+		},
+	}
+}
+
+#[cfg(feature = "std")]
+pub fn build_grid(
+	submitted: Vec<AppExtrinsic>,
+	block_length: BlockLength,
+	seed: Seed,
+) -> Result<EvaluationGrid, String> {
+	let grid = EvaluationGrid::from_extrinsics(
+		submitted.to_vec(),
```

Comment:
Is `.to_vec()` needed?

Reply:
Oh yeah, it was not a Vec at the time I did it the first time